apps/preview/next/pages/showcases/SelectDropdown/SelectDropdownError.tsx

```diff
@@ -0,0 +1,121 @@
+import { ShowcasePageLayout } from '../../showcases';
+// #region source
+import { useId, useRef, useState, type KeyboardEvent } from 'react';
+import classNames from 'classnames';
+import { SfIconExpandMore, SfListItem, SfDropdown, SfIconCheck, useTrapFocus } from '@storefront-ui/react';
+
+type OptionType = {
+  label: string;
+  value: string;
+};
+
+const options: OptionType[] = [
+  {
+    label: 'Today',
+    value: 'today',
+  },
+  {
+    label: 'Tomorrow',
+    value: 'tomorrow',
+  },
+  {
+    label: 'Anytime',
+    value: 'anytime',
+  },
+];
+
+export default function SelectDropdownWithPlaceholder() {
+  const [listVisible, setListVisible] = useState(false);
+  const [selectedOption, setSelectedOption] = useState<OptionType>();
+  const listboxId = `select-dropdown-${useId()}`;
+
+  const selectDropdownRef = useRef<HTMLDivElement>(null);
+  const listboxRef = useRef<HTMLUListElement>(null);
+
+  useTrapFocus(listboxRef, {
+    arrowKeysOn: true,
+    activeState: listVisible,
+    initialFocusContainerFallback: true,
+  });
+
+  const selectOption = (option: OptionType) => {
+    setSelectedOption(option);
+    setListVisible(false);
+    selectDropdownRef.current?.focus();
+  };
+
+  const handleTriggerKeyDown = (e: KeyboardEvent<HTMLDivElement>) => {
+    if (e.key === ' ') setListVisible((isOpen) => !isOpen);
+  };
+
+  const handleOptionItemKeyDown = (e: KeyboardEvent<HTMLLIElement>, option: OptionType) => {
+    if (e.key === ' ' || e.key === 'Enter') selectOption(option);
+  };
+
+  return (
+    <>
+      <SfDropdown
+        open={listVisible}
+        onClose={() => setListVisible(false)}
+        className="relative !w-full"
+        dropdownClassName="w-full"
+        trigger={
+          <label className="font-medium typography-label-sm">
+            Speed
+            <div
+              ref={selectDropdownRef}
+              role="combobox"
+              aria-controls={listboxId}
+              aria-expanded={listVisible}
+              aria-label="Select one option"
+              aria-activedescendant={selectedOption ? `${listboxId}-${selectedOption.value}` : undefined}
+              className={classNames(
+                'mt-0.5 flex items-center gap-8 relative font-normal typography-text-base ring-inset rounded-md py-2 px-4 cursor-pointer',
+                selectedOption
```

Comment:
Could you make this a bit more generic and create `isValid` variable like you did with `isDisabled` in the previous example? Then use `isValid` here. Then somebody how copy this, could easily change their condition under `isValid`.

Reply:
2a7f6108b1c05b6b7a73c0534d9993adca8ba622